src/zc/buildout/testing.py

```diff
@@ -110,12 +110,20 @@ def clean_up_pyc(*path):
 MUST_CLOSE_FDS = not sys.platform.startswith('win')
 
 def system(command, input='', with_exit_code=False):
+    env = dict(os.environ)
+    if 'xterm' in env.get('TERM', ''):
+        # xterm and its variants add invisible control characters,
+        # which we do not want as they mess up doctests.  See:
+        # https://github.com/buildout/buildout/pull/311
+        # http://bugs.python.org/issue19884
+        env['TERM'] = 'dumb'
```

Comment:
What do you think about always setting it to `dumb` regardless of the original terminal?

To me it would be simpler and less surprising. I keep imagining someone else a few months or years from now pulling their hair out trying to figure out why subprocesses of buildout tests only get `dumb` terminals under some circumstances but not others...

Besides, a line like this feels oddly satisfying:

    env = dict(os.environ, TERM='dumb')